.github/workflows/release.yml

```diff
@@ -1,100 +1,77 @@
+              uses: crazy-max/ghaction-import-gpg@v6
+              with:
+                  gpg_private_key: ${{ secrets.GPG_PRIVATE_KEY }}
+                  passphrase: ${{ secrets.GPG_PASSPHRASE }}
+                  fingerprint: ${{ env.GPG_FINGERPRINT }}
 
-      - name: Release
-        env:
-          GITHUB_TOKEN: ${{ secrets.REPOSITORY_PUSH_ACCESS }}
-        run: npm exec semantic-release
+            - name: Release
+              env:
+                  GITHUB_TOKEN: ${{ secrets.REPOSITORY_PUSH_ACCESS }}
+              run: npm exec semantic-release
 
-      - name: Setup QEMU
-        uses: docker/setup-qemu-action@v3
+            - name: Set Portainer stack webhook URL based on branch
+              run: |
+                  if [[ "${{ github.ref }}" == "refs/heads/main" ]]; then
+                    PORTAINER_WEBHOOK_URL=${{ secrets.PORTAINER_WEBHOOK_MAIN_URL }}
+                  else
+                    PORTAINER_WEBHOOK_URL=${{ secrets.PORTAINER_WEBHOOK_DEV_URL }}
+                  fi
+                  echo "PORTAINER_WEBHOOK_URL=$PORTAINER_WEBHOOK_URL" >> $GITHUB_ENV
+            - name: Trigger Portainer stack update
+              uses: newarifrh/portainer-service-webhook@v1
+              with:
+                  webhook_url: ${{ env.PORTAINER_WEBHOOK_URL }}
 
-      - name: Extract metadata for the Docker image
-        id: metadata
-        uses: docker/metadata-action@v5
-        with:
-          images: ghcr.io/${{ env.IMAGE_NAME }}
-          flavor: |
-            latest=${{ startsWith(github.ref, 'refs/heads/main') }}
-            suffix=-${{ github.sha }}
-
-      - name: Login to GitHub Container Registry
-        uses: docker/login-action@v3
-        with:
-          registry: ghcr.io
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
-
-      - name: Build and push Docker image
-        id: build
-        uses: docker/build-push-action@v5
-        with:
-          platforms: linux/amd64,linux/arm64
-          cache-from: type=gha
-          cache-to: type=gha,mode=max,ignore-error=true
-          push: true
-          tags: ${{ steps.metadata.outputs.tags }}
-          labels: ${{ steps.metadata.outputs.labels }}
-          build-args: |
-            GITHUB_ACTOR=${{ github.actor }}
-            GITHUB_TOKEN=${{ secrets.GITHUB_TOKEN }}
-
-      - name: Set Portainer stack webhook URL based on branch
-        run: |
-          if [[ "${{ github.ref }}" == "refs/heads/main" ]]; then
-            PORTAINER_WEBHOOK_URL=${{ secrets.PORTAINER_WEBHOOK_MAIN_URL }}
-          else
-            PORTAINER_WEBHOOK_URL=${{ secrets.PORTAINER_WEBHOOK_DEV_URL }}
-          fi
-          echo "PORTAINER_WEBHOOK_URL=$PORTAINER_WEBHOOK_URL" >> $GITHUB_ENV
-
-      - name: Trigger Portainer stack update
-        uses: newarifrh/portainer-service-webhook@v1
-        with:
-          webhook_url: ${{ env.PORTAINER_WEBHOOK_URL }}
+            - name: Purge outdated images
```

Comment:
This should be its own workflow. It doesn't belong in a "Release" workflow.

Reply:
Running it on its own can be a waste of resources, considering that this is a stable API, there may not be releases that often

We will see long periods of time with no updates, the workflow will run unnecessarily multiple times between releases, and not do anything because we keep the latest x releases anyways, running after a release is the best time to run